command.lisp

```diff
@@ -338,9 +338,18 @@ then describes the symbol."
       ,@body)))
 
 (define-stumpwm-type :y-or-n (input prompt)
-  (let ((s (or (argument-pop input)
-               (read-one-line (current-screen) (concat prompt "(y/n): ")))))
-    (equal s "y")))
+  (let* ((valid '(("y" . t)
+                  ("Y" . t)
+                  (t . t)
+                  ("n" . nil)
+                  ("N" . nil)
+                  (nil . nil)))
+         (s (or (argument-pop input)
+                (read-one-line (current-screen) (concat prompt "(y/n): "))))
+         (out (assoc s valid :test 'equal)))
```

Comment:
There is no need for an assoc list. Just

```(let ((s (or (argument-pop input)
               (read-one-line (current-screen) (concat prompt "(y/n): ")))))
    (member s '("y" t) :test #'equal)))
```

In Lisp the expection for predicates is nil => false, anything else => true.

Reply:
> Is it going to be a problem that this function can now throw an error? It adds the potential for a non-local exit everywhere that it is called, although I would hope that anything important enough to be affected would already be wrapped in `unwind-protect` or `handler-case`.

In my testing, it's manifested itself as a message displayed when i try certain commands (with invalid arguments), although that's probably coming from a `handler-case` somewhere

maybe expanding the "affirmative" values is enough and we don't need to be strict about the negative ones -- and thanks for the advice about `member`, i'm still learning